system/jhtree/jhtree.cpp

```diff
@@ -992,6 +992,12 @@ void CKeyIndex::init(KeyHdr &hdr, bool isTLK, bool allowPreload)
             }
         }
         rootNode = nodeCache->getNode(this, iD, rootPos, NULL, isTLK);
+        if (rootNode && isTLK && !rootNode->isLeaf() && rootNode->getNumKeys()==1)   // In theory should work for non-TLK keys too, but not been tested and less likely to happen
```

Comment:
Could also do with a comment similar to the commit text.

Reply:
added